constraint/core.go

```diff
@@ -240,20 +241,27 @@ func (system *System) AddSecretVariable(name string) (idx int) {
 	return idx
 }
 
-func (system *System) AddSolverHint(f solver.Hint, input []LinearExpression, nbOutput int) (internalVariables []int, err error) {
+func (system *System) AddSolverHint(f solver.Hint, id solver.HintID, input []LinearExpression, nbOutput int) (internalVariables []int, err error) {
 	if nbOutput <= 0 {
 		return nil, fmt.Errorf("hint function must return at least one output")
 	}
 
+	// TODO @Tabaie @gbotrel consider getting rid of hint names entirely
```

Comment:
names are good to derive human readable error messages if hints are missing or duplicate, etc ?